CI fails with dev dependencies: AttributeError: 'Qwen3VLTextConfig' object has no attribute 'pad_token_id'

https://github.com/huggingface/trl/actions/runs/21066816117/job/60586076404
> AttributeError: 'Qwen3VLTextConfig' object has no attribute 'pad_token_id'. Did you mean: 'bos_token_id'?
```python
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_vlm[trl-internal-testing/tiny-Qwen3VLForConditionalGeneration] - AttributeError: 'Qwen3VLTextConfig' object has no attribute 'pad_token_id'. Did you mean: 'bos_token_id'?
  FAILED tests/test_utils.py::TestForwardMaskedLogits::test_vlm[trl-internal-testing/tiny-Qwen3VLForConditionalGeneration] - AttributeError: 'Qwen3VLTextConfig' object has no attribute 'pad_token_id'. Did you mean: 'bos_token_id'?
```
Stacktrace:
```python
  >       model = AutoModelForImageTextToText.from_pretrained(model_id, dtype="auto", device_map=device)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  
  tests/test_utils.py:927: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  .venv/lib/python3.12/site-packages/transformers/models/auto/auto_factory.py:372: in from_pretrained
      return model_class.from_pretrained(
  .venv/lib/python3.12/site-packages/transformers/modeling_utils.py:4020: in from_pretrained
      model = cls(config, *model_args, **model_kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/models/qwen3_vl/modeling_qwen3_vl.py:1289: in __init__
      self.model = Qwen3VLModel(config)
                   ^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/models/qwen3_vl/modeling_qwen3_vl.py:938: in __init__
      self.language_model = Qwen3VLTextModel._from_config(config.text_config)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/modeling_utils.py:1464: in _from_config
      model = cls(config, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/models/qwen3_vl/modeling_qwen3_vl.py:807: in __init__
      self.padding_idx = config.pad_token_id
                         ^^^^^^^^^^^^^^^^^^^
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  self = Qwen3VLTextConfig {
    "attention_bias": false,
    "attention_dropout": 0.0,
    "bos_token_id": 151643,
    "dtype": "bfloa...,
    "tie_word_embeddings": true,
    "transformers_version": "5.0.0.dev0",
    "use_cache": true,
    "vocab_size": 151936
  }
  
  key = 'pad_token_id'
  
      def __getattribute__(self, key):
          if key != "attribute_map" and key in super().__getattribute__("attribute_map"):
              key = super().__getattribute__("attribute_map")[key]
  >       return super().__getattribute__(key)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  E       AttributeError: 'Qwen3VLTextConfig' object has no attribute 'pad_token_id'. Did you mean: 'bos_token_id'?
  
  .venv/lib/python3.12/site-packages/transformers/configuration_utils.py:164: AttributeError
```

--- a/tests/test_sft_trainer.py
+++ b/tests/test_sft_trainer.py
@@ -1357,10 +1357,16 @@ class TestSFTTrainer(TrlTestCase):
             # "trl-internal-testing/tiny-SmolVLMForConditionalGeneration", seems not to support bf16 properly
             pytest.param(
                 "trl-internal-testing/tiny-Qwen3VLForConditionalGeneration",
-                marks=pytest.mark.skipif(
-                    Version(transformers.__version__) < Version("4.57.0"),
-                    reason="Qwen3-VL series were introduced in transformers-4.57.0",
-                ),
+                marks=[
+                    pytest.mark.skipif(
+                        Version(transformers.__version__) < Version("4.57.0"),
+                        reason="Qwen3-VL series were introduced in transformers-4.57.0",
+                    ),
+                    pytest.mark.xfail(
+                        Version(transformers.__version__) >= Version("5.0.0.dev0"),
+                        reason="Blocked by upstream transformers bug (transformers#43334)",
+                    ),
+                ],
             ),
         ],
     )

--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -915,10 +915,16 @@ class TestForwardMaskedLogits:
             # "trl-internal-testing/tiny-SmolVLMForConditionalGeneration", seems not to support bf16 properly
             pytest.param(
                 "trl-internal-testing/tiny-Qwen3VLForConditionalGeneration",
-                marks=pytest.mark.skipif(
-                    Version(transformers.__version__) < Version("4.57.0"),
-                    reason="Qwen3-VL series were introduced in transformers-4.57.0",
-                ),
+                marks=[
+                    pytest.mark.skipif(
+                        Version(transformers.__version__) < Version("4.57.0"),
+                        reason="Qwen3-VL series were introduced in transformers-4.57.0",
+                    ),
+                    pytest.mark.xfail(
+                        Version(transformers.__version__) >= Version("5.0.0.dev0"),
+                        reason="Blocked by upstream transformers bug (transformers#43334)",
+                    ),
+                ],
             ),
         ],
     )

